library/cwm/test/common_widgets_test.rb

```diff
@@ -59,3 +58,47 @@ def hspacing
     end
   end
 end
+
+describe CWM::RichText do
+  subject { described_class.new }
+  let(:widget_id) { Id(subject.widget_id) }
+
+  describe "#value=" do
+    before do
+      allow(subject).to receive(:keep_scroll?).and_return(keep_scroll)
+      allow(Yast::UI).to receive(:ChangeWidget)
+    end
+
+    context "when set to restore the scroll" do
+      let(:keep_scroll) { true }
+
+      it "saves the scroll position" do
+        expect(Yast::UI).to receive(:QueryWidget).with(widget_id, :VScrollValue)
+
+        subject.value = "Test"
+      end
+
+      it "restores the scroll" do
+        expect(Yast::UI).to receive(:ChangeWidget).with(widget_id, :VScrollValue, anything)
+
+        subject.value = "Test"
+      end
```

Comment:
well, it saves, but it cannot be used outside. So if we use rspec as behavior documentation that I expect something like `keeps scrolling values` but it is also not precise as we remember only vertical scrolling and not horizontal right?

And what is missing here is mentioning original behavior that is kept, so it sets value of widget.

Reply:
> vertical scrolling and not horizontal right?

Right. At some point, I had _vertical scrolling_ there, but moving/changing things around...

> And what is missing here is mentioning original behavior that is kept, so it sets value of widget.

Also true. But I was waiting to collect enough feedback to know what we want to do: preserve the original behavior or change it as @lslezak proposed.

Reply:
> 
> > And what is missing here is mentioning original behavior that is kept, so it sets value of widget.
> 
> Also true. But I was waiting to collect enough feedback to know what we want to do: preserve the original behavior or change it as @lslezak proposed.

It is probably misunderstanding, I mean here that we do not test that we change content of richtext which is the most important behavior of value= method. And we should do it regardless if scroll saving is set or not.